### ISSUE TITLE
Flatpak Support

### ISSUE THREAD

**Issue by Doomsdayrs** [intiface/intiface-central#62]
Publication to  [flathub](https://flathub.org/home) via flatpak for Linux users.

## Nice to haves:
- [x] Upstream metainfo #124
- [x] Upstream desktop file #123
- [x] Upstream svg icon #97 
- [ ] Build instructions
- [ ] Use https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html for configs & logs

## Issues
- [ ] Upstream license clarification? Since it is a dual license and I do not know the SPDX for the commercial. https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-project_license
- [ ] #40 
Is causing issues when trying to load from sandbox.

Current progress: https://github.com/Doomsdayrs/flathub/tree/com.nonpolynomial.intiface_central

**Comment by Doomsdayrs** [comment]
It would nice to have an SVG version of the icon

**Comment by Doomsdayrs** [comment]
Good news, Past me had made this work for v2.3.0 under flatpak

Bad news, v2.4.3 hangs at loading screen.

@qdot Is there anyway to debug the compiled application?

**Comment by Doomsdayrs** [comment]
speaking of, do you have the svg of the icon @qdot ?

**Comment by qdot** [comment]
For the hang: Wait for 2.4.4. I'm guessing this has to do with the openssl version, which I'll now be vendoring into the builds. If you want to check and see if that's the case, run ldd on lib_intiface_flutter_rust_bridge or whatever that library is, see if there's a missing dependency.

As for an SVG, I'll have to look.

**Comment by Doomsdayrs** [comment]
@qdot Any update on the SVG?

**Comment by Doomsdayrs** [comment]
Testing v2.4.4, Still hangs on loading screen.

![image](https://github.com/intiface/intiface-central/assets/38189170/425704f6-70ea-452d-b620-dfd191e904e4)

**Comment by qdot** [comment]
Hmm. Wonder if this is a config file issue. Try to find wherever the intiface_central directory is (possibly somewhere in `~/.config`) and delete it, see if that makes things come up.

**Comment by Doomsdayrs** [comment]
> Hmm. Wonder if this is a config file issue. Try to find wherever the intiface_central directory is (possibly somewhere in `~/.config`) and delete it, see if that makes things come up.

Just coming right now from doing exactly that.

Wiped `~/.var/app/com.nonpolynomial.intiface_central` and restarted the app.

Issue persists.

Console output:
```
$ flatpak run com.nonpolynomial.intiface_central 

(intiface_central:2): Gdk-CRITICAL **: 22:52:39.110: gdk_window_get_state: assertion 'GDK_IS_WINDOW (window)' failed

** (intiface_central:2): WARNING **: 22:53:04.144: atk-bridge: get_device_events_reply: unknown signature
```

Log content:
```
[I] Global Loggy: Sentry URL set, crash and log reporting available.
[I] Global Loggy: /home/doomsdayrs/.var/app/com.nonpolynomial.intiface_central/data/intiface_central/logs
[I] Global Loggy: Device configuration file does not exist, returning 0.0.
[I] Global Loggy: {}
[I] Global Loggy: []
[E] Global Loggy: Error loading cubit! Deleting configs and creating new ones.
```

**Comment by rugk** [comment]
I agree it would really be nice if you could publish this as a [flatpak](https://flatpak.org/) on [flathub](https://flathub.org/) e.g.

Flatpaks are a new software distribution mechanism for Linux distros, can thus installed on any distro and are easy to update. They are easy to install _and_ update and work on all Linux distros.
Also, if you publish it on _FlatHub_ you may grow your user base given many distros include that and it is a common software source, so your app can be discovered more easily. 

Here is [how to get started](http://docs.flatpak.org/en/latest/getting-started.html).

**Comment by ToasterUwU** [comment]
Anyway i could help? 

Linux User here, so im very interested in getting this going. A binary to manually copy around is very duct tape and hope style, i rather have something proper. So if there is anything you need help with, im happy to help.

**Comment by ToasterUwU** [comment]
@Doomsdayrs Can i help you out on this somehow? Im also more than happy to carrry the torch from here if you are not available for this anymore. Just hand everything you have on this and i will take it from here if you want me to

**Comment by Doomsdayrs** [comment]
Sorry for the long wait for replies. I shall do a re-evaluation of this shortly.

**Comment by xiro49-dev** [comment]
any news ?

**Comment by rugk** [comment]
@xiro49-dev Well silently, a PR for inclusion on Flathub has been proposed hehe… (one does not get a notification for that) if you look above your commment: https://github.com/flathub/flathub/pull/5012

There you can see the latest things.

**Comment by Doomsdayrs** [comment]
OH GOSH ITS BEEN 4 MONTHS? I'll most certainly do this next week when I have time. 😭

**Comment by Doomsdayrs** [comment]
> [@Doomsdayrs](https://github.com/Doomsdayrs) I was too embarrassed to ping you again lol, was so close to packing it myself for nixpkgs

Don't be embarrassed, I have the memory of a fish. >.<)!

But yes, I think sunday evening time for me I'll put all my focus I can into this.

**Comment by Doomsdayrs** [comment]
Alright may not be sunday but this will be the time

**Comment by Doomsdayrs** [comment]
Two years comes to close soon 💜

**Comment by xiro49-dev** [comment]
@Doomsdayrs thank you so much cant wait

**Comment by qdot** [comment]
[Done.](https://flathub.org/apps/com.nonpolynomial.intiface_central)
